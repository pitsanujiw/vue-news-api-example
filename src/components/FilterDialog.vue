<template>
  <div class="d-flex justify-end">
    <v-dialog v-model="dialog" width="500">
      <template v-slot:activator="{ on, attrs }">
        <v-btn color="primary" dark v-bind="attrs" v-on="on"> Filter </v-btn>
      </template>

      <v-card>
        <v-card-title class="text-h5 grey lighten-2">
          FILTER NEWS
        </v-card-title>

        <v-card-text>
          <v-text-field
            label="Search news"
            placeholder="Search"
            :value="getSearch"
            @change="setSearch"
          ></v-text-field>
          <v-autocomplete
            :items="getNewsSources"
            color="white"
            item-text="name"
            label="sources"
            item-value="id"
            @change="setSource"
            :value="getSources"
            clearable
          ></v-autocomplete>
        </v-card-text>

        <v-divider></v-divider>

        <v-card-actions>
          <v-btn depressed @click="onClose()">CLOSE</v-btn>
          <v-spacer></v-spacer>
          <v-btn color="primary" depressed @click="onFilter()">SEARCH</v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </div>
</template>

<script>
import { mapGetters, mapMutations, mapActions } from 'vuex';

import { SET_QUERY, SET_SOURCES_FILTER } from '@/constants';

export default {
  name: 'FilterDialog',
  data: () => ({
    dialog: false,
  }),
  watch: {},
  methods: {
    ...mapMutations({
      setQuery: SET_QUERY,
      setSourcesText: SET_SOURCES_FILTER,
    }),
    ...mapActions(['newsTopHighlight']),
    setSearch(text) {
      this.setQuery(text);
    },
    onClose() {
      this.dialog = false;
    },
    setSource(source) {
      this.setSourcesText(source);
    },
    async onFilter() {
      this.dialog = false;
      await this.newsTopHighlight(true);
    },
  },
  computed: {
    ...mapGetters(['getSearch', 'getNewsSources', 'getSources']),
  },
};
</script>

<style>
</style>
