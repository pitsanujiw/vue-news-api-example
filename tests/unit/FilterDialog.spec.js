import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import FilterDialog from '../../src/components/FilterDialog.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('FilterDialog', () => {
  const modules = {
    getters: { getNewsSources: jest.fn(), geSources: jest.fn(), getSearch: jest.fn() },
  };
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
    store = new Vuex.Store({
      modules: {
        newModules: modules,
      },
    });
  });

  it('Renders FilterDialog Component', () => {
    const wrapper = shallowMount(FilterDialog, { store, localVue, vuetify });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
