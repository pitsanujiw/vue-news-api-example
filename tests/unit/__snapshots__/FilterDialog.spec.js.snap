// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDialog Renders FilterDialog Component 1`] = `
<div class="d-flex justify-end">
  <v-dialog-stub opendelay="0" closedelay="0" contentclass="" origin="center center" retainfocus="true" transition="dialog-transition" width="500">
    <v-card-stub loaderheight="4" tag="div">
      <v-card-title-stub tag="div" class="text-h5">
        Search news top highlight
      </v-card-title-stub>
      <v-card-text-stub tag="div">
        <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Search news" loaderheight="2" clearicon="$clear" placeholder="Search" type="text"></v-text-field-stub>
        <v-autocomplete-stub color="white" errorcount="1" errormessages="" messages="" rules="" successmessages="" appendicon="$dropdown" backgroundcolor="" label="sources" loaderheight="2" clearable="true" clearicon="$clear" type="text" valuecomparator="[Function]" nodatatext="$vuetify.noDataText" items="" itemcolor="primary" itemdisabled="disabled" itemtext="name" itemvalue="id" menuprops="[object Object]" allowoverflow="true" filter="[Function]"></v-autocomplete-stub>
      </v-card-text-stub>
      <v-divider-stub></v-divider-stub>
      <v-card-actions-stub tag="div">
        <v-btn-stub tag="button" activeclass="" depressed="true" type="button">CLOSE</v-btn-stub>
        <v-spacer-stub tag="div"></v-spacer-stub>
        <v-btn-stub dark="true" tag="button" activeclass="" type="button">SEARCH</v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
</div>
`;
